MediaBrowser.Common/Extensions/HttpContextExtensions.cs

```diff
@@ -15,8 +15,8 @@ public static class HttpContextExtensions
         /// <returns><c>true</c> if the request is coming from LAN, <c>false</c> otherwise.</returns>
         public static bool IsLocal(this HttpContext context)
         {
-            return (context.Connection.LocalIpAddress == null
-                    && context.Connection.RemoteIpAddress == null)
+            return context.Connection.LocalIpAddress == null
+                   || context.Connection.RemoteIpAddress == null
```

Comment:
I've been trying to figure out when these are null, but it's not easy. I think this particular line should be removed though. Either Local IP is null (unix socket) or Local IP and Remote IP are the same. I don't see a situation where Local IP is not null and Remote IP _is_, but Remote can be null in the case where x-forwarded-for isn't used.

I dunno if any of this makes sense, but these are my thoughts.

Reply:
Yeah, i mentioned this in the description. I think the remote being null implies the local bring null. I simply left this check in to be a bit more defensive. However since then the code was changed to use the Equals funtion instead of a method anyways so it shouldn't crash if remote is null. So the defensiveness is less important.

So let me know what you think. We can definitely remove this line and add it back if we find a case where remote is null and local isn't.

Reply:
`IsLocal` isn't used for much, so I think it's fine to remove that particular check.

Reply:
Actually I think I misunderstood. This line can't be removed because for a Unix socket connection with `X-Forwarded-For` the Local IP address will be null but the remote socket will be set based on `X-Forwarded-For`. I think we can remove the local address check only. See the suggestion I have added, the logic is then very straightforward.

The request is considered local if either
- The remote address is null (unix socket)
- The remote address matches the local address.

That being said this whole check is sort of suspect. The docs say "if the request is coming from LAN" but this only checks for a request from the same machine. But this PR is probably not the place to extend this to check to the LAN.

Reply:
I've pushed a new version. There was in fact a bug in the previous version because it considered all requests as Local if you were listening on a UNIX socket. This isn't completely wrong but it is better to handle `X-Forwarded-For` like is done for IP addresses. This change now treats a null remote as a local request (UNIX socket without `X-Forwarded-For`) otherwise always does the IP check.

Reply:
Based on your findings, isn't it local when "Local == Remote" and everything else is irrelevant?

1. For a unix socket the local address is null and without x-forwarded-for the remote address is null.
2. For HTTP requests local is 127.0.0.1 or localhost and remote must match.
3. None of this works if there's a misconfigured reverse proxy in front

Reply:
Yes you are correct (assuming that the `Equals` function handles null correctly). However I thought this was too subtle and likely to regress in the future. If you want I can add a comment instead but it seems a lot harder to explain that the `null == null` case is a unix socket and must be preserved than simply adding a separate clause to handle that case and make it clear that the second case only needs to handle IP.